solnlib/bulletin_rest_client.py

```diff
@@ -0,0 +1,104 @@
+import logging
+from solnlib import splunk_rest_client as rest_client
+from splunklib import binding
+import json
+
+
+__all__ = ["BulletinRestClient"]
+
+
+class BulletinRestClient:
+    READ_ENDPOINT = "/services/admin/messages"
+    CREATE_DELETE_ENDPOINT = "/services/messages"
+
+    headers = [("Content-Type", "application/json")]
+
+    def __init__(
+        self,
+        message_name: str,
+        session_key: str,
+        logger: logging.Logger = None,
+        **context: dict,
+    ):
+
+        self.message_name = message_name
+        self.session_key = session_key
+
+        if logger:
+            self.logger = logger
+        else:
+            self.logger = logging
+
+        self._rest_client = rest_client.SplunkRestClient(
+            self.session_key, app="-", **context
+        )
+        self.logger.info(f"mymy123 init {self.message_name}")
+
+    def create_message(self, msg):
+        body = {"name": self.message_name, "value": msg}
+        self.logger.info(f"mymy123 create_message {self.message_name} {msg}")
+        try:
+            self._rest_client.post(
+                self.CREATE_DELETE_ENDPOINT, body=body, headers=self.headers
+            )
+        except binding.HTTPError as e:
+            if 400 <= e.status < 500:
+                self.logger.error(
+                    f"Create message failed - {e.status} "
+                    f"Client Error: {e.reason} for url: {self.CREATE_DELETE_ENDPOINT}"
+                )
+            if 500 <= e.status < 600:
```

Comment:
Shouldn't this be an if...elif block instead of multiple ifs?

Reply:
yes, sorry for the confusion but I had a git problem this morning and it pushed only my first, working commit. At the moment I have fixed the issue with Git and this is a PR that should be reviewed.